float8_experimental/float8_linear_utils.py

```diff
@@ -291,11 +291,10 @@ def inner_func():
         ), "Mismatched lengths of amax tensors."
 
         if dist.is_initialized():
-            # Combine all the amax tensors into one tensor and reduce it
-            # Note: do not reduce the weight values, because FSDP already ensures
-            # the weight values on all ranks are the same after all-gather.
             all_amax_tensors = torch.cat(
-                fp8_amax_x_tensor_list + fp8_amax_dL_dY_tensor_list
+                fp8_amax_x_tensor_list
+                + fp8_amax_w_tensor_list
```

Comment:
should we only do this if we are using fp8 all gather ?

Reply:
that could make sense, I'd love to see the data to see if this is going to matter for performance.  Focusing on numerics for now, was hoping for performance be tackled in future PRs.